docs(readme): explain how `it` and `_` can differ for method calls by scope (#16)
citycide/param.macro: readme.md

array.map(_it => _it)
```

> Likewise, only at the top-level and as the right-hand side of an assignment does `it` and `_` behave similarly when used to produce implicit n-ary functions — since there's no further upward to go they can both stop at the same place.

Reviewer: ```suggestion
An exception to these scoping differences is at the top-level, like the right-hand
side of an assignment. `it` and `_` behave similarly here since there's no further
upward to go, so they'll both happen to target the same place.
```

> The following two map implementations do the same thing:

Reviewer: ```suggestion
For example the following two map implementations do the same thing:
```

```js
import { _, it } from 'param.macro'

const map1 = _.map(_)
const map2 = it.map(_)
```

> _However_, if nested deeper inside a function call, the object placeholder `_` would traverse upward and create a separate function _first_, before being passed the `_` inside the method call itself. This creates a unary method call instead of the implicit binary function we wanted, lift or not.

Reviewer: ```suggestion
_However_, if nested deeper inside a function call the object placeholder `_` in
`map1` above would traverse further upward than an `it` would, and create a separate
function _first_, before the argument placeholder `_` inside the method call itself.
This creates an unary method call instead of the implicit binary function we probably
wanted, `lift` or not.
```

> The `it` implementation _does_ still create the implicit binary function, even if nested deeper. And following their own rules any `_` inside the method call will traverse up to the method call and stop to create a function there, as we wanted.

Reviewer: ```suggestion
The `it` implementation in `map2` above _does_ still create the implicit binary
function, even if nested deeper. And following the normal placeholder rules, any `_`
inside the method call will traverse up to the method call and stop to create a
function there, as we wanted.
```

Author: Oops. Hasn't noticed the line length at all.
